pkg/operator/ceph/cluster/mgr/dashboard.go

```diff
@@ -201,26 +203,41 @@ func (c *Cluster) setLoginCredentials(password string) error {
 	// Set the login credentials. Write the command/args to the debug log so we don't write the password by default to the log.
 	logger.Infof("setting ceph dashboard %q login creds", dashboardUsername)
 
+	// Generate a temp file
+	file, err := ioutil.TempFile("", "")
+	if err != nil {
+		return errors.Wrap(err, "failed to generate temp file")
+	}
+	err = ioutil.WriteFile(file.Name(), []byte(password), 0444)
+	if err != nil {
+		return errors.Wrap(err, "failed to write into file")
+	}
 	// retry a few times in the case that the mgr module is not ready to accept commands
-	_, err := client.ExecuteCephCommandWithRetry(func() (string, []byte, error) {
-		args := []string{"dashboard", "set-login-credentials", dashboardUsername, password}
+	_, err = client.ExecuteCephCommandWithRetry(func() (string, []byte, error) {
+		args := []string{"dashboard", "ac-user-create", dashboardUsername, "-i", file.Name(), "administrator"}
 		cmd := client.NewCephCommand(c.context, c.clusterInfo, args)
 		output, err := cmd.RunWithTimeout(client.CmdExecuteTimeout)
 		return "set dashboard creds", output, err
 	}, c.exitCode, 5, invalidArgErrorCode, dashboardInitWaitTime)
 	if err != nil {
-		return errors.Wrapf(err, "failed to set login creds on mgr")
+		return errors.Wrap(err, "failed to set login creds on mgr")
 	}
+	defer func() error {
```

Comment:
No need to return an error, we can just log it and continue
```suggestion
	defer func() {
```

Reply:
Done.